contracts/contracts/SimpleReadAccessController/library.cairo

```diff
@@ -0,0 +1,57 @@
+%lang starknet
+
+from starkware.cairo.common.alloc import alloc
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from starkware.starknet.common.syscalls import get_tx_info
+from starkware.cairo.common.bool import TRUE, FALSE
+
+from SimpleWriteAccessController.library import simple_write_access_controller
+
+namespace simple_read_access_controller:
+    func constructor{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
```

Comment:
Didn't you mention that contracts with constructors are not composable, so we restructured them, but now this library also has a constructor. 

Asking questions here, not sure how to best structure this.

Reply:
Theoretical preambule:

- For cairo to perceives func as constructor when it needs to have [`@constructor` decorator + func name is constructor](https://github.com/starkware-libs/cairo-lang/blob/167b28bcd940fd25ea3816204fa882a0b0a49603/src/starkware/starknet/compiler/external_wrapper.py#L212)
- Functions with decorators can't be used in `namespace`s
- `namespace` name prepends to functions name + [some magic static string values](https://github.com/starkware-libs/cairo-lang/blob/167b28bcd940fd25ea3816204fa882a0b0a49603/src/starkware/starknet/compiler/external_wrapper.py#L723)

That means if file contains constructor (according to definition above) and then includes file that contains constructor we will have `redefinition` error. So what we do to avoid this is restructure code to have `library.cairo` & `some_contract.cairo`. Then use namespace trickery to avoid naming functions smth like that `simple_read_access_controller_constructor`

Reply:
> Theoretical preambule:
> 
> * For cairo to perceives func as constructor when it needs to have [`@constructor` decorator + func name is constructor](https://github.com/starkware-libs/cairo-lang/blob/167b28bcd940fd25ea3816204fa882a0b0a49603/src/starkware/starknet/compiler/external_wrapper.py#L212)
> * Functions with decorators can't be used in `namespace`s
> * `namespace`'s name prepends to functions name + [some magic static string values](https://github.com/starkware-libs/cairo-lang/blob/167b28bcd940fd25ea3816204fa882a0b0a49603/src/starkware/starknet/compiler/external_wrapper.py#L723)
> 
> That means that if file contains constructor (according to definition above) and then includes file that contains constructor we will have `redefinition` error because we can't move it to namespace. So what we do to avoid this is restructure code to have `library.cairo` & `some_contract.cairo`. Then use namespace trickery to avoid naming functions smth like that `simple_read_access_controller_constructor`

Reply:
What do we do with this one? :)

Reply:
Rename to `initializer`

Reply:
Done. Renamed to initialise as suggested in slack dos